src/wazuh_modules/agent_upgrade/wm_agent_upgrade.h

```diff
@@ -0,0 +1,80 @@
+/*
+ * Wazuh Module for Agent Upgrading
+ * Copyright (C) 2015-2020, Wazuh Inc.
+ * July 15, 2020.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#ifndef WM_AGENT_UPGRADE_H
+#define WM_AGENT_UPGRADE_H
+
+#include "wazuh_modules/wmodules.h"
+
+int wm_agent_upgrade_read(xml_node **nodes, wmodule *module);
+
+extern const wm_context WM_AGENT_UPGRADE_CONTEXT;   // Context
+
+cJSON *wm_agent_process_upgrade_command(const cJSON* params, const cJSON* agents);
+cJSON* wm_agent_process_upgrade_result_command(const cJSON* agents);
+/**
+ * Module general configuration
+ * */
+
+typedef enum _wm_upgrade_state {
+    NOT_STARTED,
+    STARTED,
+    ERROR
+} wm_upgrade_state;
+
+typedef enum _wm_upgrade_error_codes {
+    SUCCESS = 0,
+    PARSING_ERROR,
+    TASK_CONFIGURATIONS,
+    TASK_MANAGER_COMMUNICATION,
+    TASK_MANAGER_FAILURE,
+    UPGRADE_ALREADY_ON_PROGRESS,
+    AGENT_ID_ERROR,
+    UNKNOWN_ERROR
+} wm_upgrade_error_codes;
+
+typedef struct _wm_agent_upgrade {
+    int enabled; ///< Indicates if modules is enabled
+} wm_agent_upgrade;
+
+/**
+ * Definition of the structure that will represent an agent doing a certain task
+ * */
+typedef struct _wm_agent_task {
+    int agent;                   ///> agent_id to be upgraded
```

Comment:
I think that the `agent` is not necessary since it will be the key of the hash table.
Also, I would name this structure just `wm_task`.

Reply:
Agreed